contexts/PaginationContext.tsx

```diff
@@ -0,0 +1,82 @@
+import { createContext, ReactNode, useState } from "react";
+import { PageStatesT, PaginateForT } from "types";
+
+export interface PaginationContextState {
```

Comment:
this context controls pagination for all pageable data.

Reply:
this is a cool approach